pylint/config/utils.py

```diff
@@ -7,38 +7,54 @@
 
 from typing import Any, Dict, Union
 
-from pylint.config.argument import _Argument, _StoreTrueArgument
-
-IMPLEMENTED_OPTDICT_KEYS = {"action", "default", "type", "choices", "help", "metavar"}
-"""This is used to track our progress on accepting all optdict keys."""
+from pylint.config.argument import _Argument, _CallableArgument, _StoreTrueArgument
+from pylint.config.callback_actions import _CallbackAction
 
 
 def _convert_option_to_argument(
     opt: str, optdict: Dict[str, Any]
-) -> Union[_Argument, _StoreTrueArgument]:
+) -> Union[_Argument, _StoreTrueArgument, _CallableArgument]:
     """Convert an optdict to an Argument class instance."""
-    # See if the optdict contains any keys we don't yet implement
     # pylint: disable-next=fixme
-    # TODO: This should be removed once the migration to argparse is finished
-    for key, value in optdict.items():
-        if key not in IMPLEMENTED_OPTDICT_KEYS:
-            print("Unhandled key found in Argument creation:", key)  # pragma: no cover
-            print("It's value is:", value)  # pragma: no cover
+    # TODO: Do something with the 'group', 'level' and 'hide' keys of optdicts
+
+    # pylint: disable-next=fixme
+    # TODO: Do something with the 'dest' key and deprecation of options
 
+    # Get the long and short flags
+    flags = [f"--{opt}"]
+    if "short" in optdict:
+        flags += [f"-{optdict['short']}"]
+
+    # Get the action type
     action = optdict.get("action", "store")
+
+    # pylint: disable-next=fixme
+    # TODO: Remove this handling after we have deprecated multiple-choice arguments
```

Comment:
`multiple-choice` values like `confidence` are not allowed in `argparse`. You can only have a `choice` or nothing.

Our transformer does the same as `multiple-choice` does, but `argparse` trips up as soon as there is a `choices` attribute. I think we should deprecate `multiple-choice`, as it is unmaintainable.

Reply:
Yeah also it makes more sense imo to use "this confidence and higher confidences" from a strictly usability point of view. Who want to get UNDEFINED and HIGH confidences, with no in between ?

Reply:
I don't even know if the feature is used at all because as you know we did not specify the confidence until very recentely.

Reply:
That would be a different, although I agreed sensible, change!

Reply:
> I don't even know if the feature is used at all because as you know we did not specify the confidence until very recentely.

I'm scared plugin writers might use it... 😢

Reply:
> I'm scared plugin writers might use it...

Yeah if it wasn't such a low level API that is probably used somewhere, we could change it to an ordered enum and get a lot of simplification. I think it would create a bigger mess than just working with what currently exists though.